### PR TITLE
Added the meta information for each event into the primary store

### LINKED ISSUE #90: Store on the primary store the meta information for each event
Store information about the `Identity` and the `Instance` for each `Event`

### PR DESCRIPTION
**Pull request by umbreak**
Fixes #90

### CODE REVIEW

**Comment by umbreak** [comment]
retest this please

**Comment by umbreak** [comment]
retest this please

File: modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/CallerCtx.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  * @param caller the author of the operation call
+  */
+final case class CallerCtx(clock: Clock, caller: Caller) {
+  lazy val meta = Meta(identity(), clock.instant())
+
+  private def identity(): Identity =
+    caller.identities.collectFirst {
+      case id: UserRef => id
+    } orElse (caller.identities.collectFirst {
+      case id: AuthenticatedRef => id
```
Comment (wwajerowicz) [inline modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/CallerCtx.scala]: shouldn't authenticated user always have `UserRef`?

Comment (umbreak) [reply modules/core/src/main/scala/ch/epfl/bluebrain/nexus/kg/core/CallerCtx.scala]: Whenever `UserRef` is present, we prefer it. If it isn't present, I try to find `AuthenticatedRef` and if it isn't present either, I set the Identity to `Anonymous`.

Now I know that by the implementation on IAM, we would never get `AuthenticatedRef` without a `UserRef`. However, that's just the implementation on IAM, which might change.

So I think that behaviour is still correct.

**Pull request merged**